src/arrays/constructors.jl

```diff
@@ -16,4 +16,16 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+function coherentstatevec(n::Int,alpha::Number;method="direct")
+    s = rand(n)
+    z = zeros(QuArray(s))
+    for i in [1:n]
+        z = z + ((alpha^(i-1))/sqrt(factorial(i)))*statevec(i, FiniteBasis(n))
+    end
+    return scale!(exp(-abs2(alpha)/2),z)
+end
+
+coherentstatevec(n::Int, alpha::Number) = displaceop(n,alpha)*statevec(1,FiniteBasis(n))
```

Comment:
I don't think we need both versions and for now I would prefer the second one. There might actually be a problem having both, because they only differ in the keywords and I am not sure how Julia handles this. You can check by calling 
```
@which coherentstatevec(20,1,method = "direct")
@which coherentstatevec(20,1)
```
To see which versions are actually called.